File: generated-sources/scalaz/mojang-api/src/main/scala/org/openapitools/client/api/OrderStatisticsResponse.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime
import OrderStatisticsResponse._

case class OrderStatisticsResponse (
  /* total amount sold */
  total: Long,
/* total sold in last 24 hours */
  last24h: Long,
/* decimal average sales per second */
  saleVelocityPerSeconds: Double)

object OrderStatisticsResponse {
  import DateTimeCodecs._

  implicit val OrderStatisticsResponseCodecJson: CodecJson[OrderStatisticsResponse] = CodecJson.derive[OrderStatisticsResponse]
  implicit val OrderStatisticsResponseDecoder: EntityDecoder[OrderStatisticsResponse] = jsonOf[OrderStatisticsResponse]
  implicit val OrderStatisticsResponseEncoder: EntityEncoder[OrderStatisticsResponse] = jsonEncoderOf[OrderStatisticsResponse]
}

File: generated-sources/dart-jaguar/mojang-api/lib/model/order_statistics_response.dart
import 'package:jaguar_serializer/jaguar_serializer.dart';

part 'order_statistics_response.jser.dart';

class OrderStatisticsResponse {
   /* total amount sold */
  @Alias('total')
  final int total;
   /* total sold in last 24 hours */
  @Alias('last24h')
  final int last24h;
   /* decimal average sales per second */
  @Alias('saleVelocityPerSeconds')
  final double saleVelocityPerSeconds;
  

  OrderStatisticsResponse(
    

{
    
     this.total = null,  
     this.last24h = null,  
     this.saleVelocityPerSeconds = null 
    }
  );

  @override
  String toString() {
    return 'OrderStatisticsResponse[total=$total, last24h=$last24h, saleVelocityPerSeconds=$saleVelocityPerSeconds, ]';
  }
}

@GenSerializer()
class OrderStatisticsResponseSerializer extends Serializer<OrderStatisticsResponse> with _$OrderStatisticsResponseSerializer {

}
